### PR TITLE
Auto: implement issue #509

### LINKED ISSUE #509: Add CopilotChat component to frontend
Subtask of #76
Depends on #508

Create `src/frontend/components/CopilotChat.tsx` with: floating button (bottom-right), collapsible chat window, message list, text input, send button. Load current customer/opportunity context from the page store and display in a context bar at the top. On send, call `POST /copilot/chat` and render tool-call results as structured cards. Suggested prompts shown on empty state. Use existing frontend patterns (API client, component structure). Acceptance: chat window opens, sends a message, and displays a response from the API.

Dev-Plan: docs/dev-plan/90-frontend/0509-add-copilotchat-component-to-frontend.md

### PR DESCRIPTION
**Pull request by yihua-zhuo**
Closes #509

Plan: `.plans/issue-509.md`

This PR was auto-implemented by Claude Code in three stages:

1. **Plan** — Claude wrote a detailed implementation plan to `.plans/issue-509.md`.
2. **Implement** — Claude executed the plan, ran `ruff check src/` + the unit and integration test suites, and iterated until all three were green.
3. **Self-review** — Claude reviewed its own diff against `codereview.md` (the project review checklist) and fixed any rule violations found, re-running the test suite if any changes were made.

PR-time CI (`Dev Agent CI`) re-verifies independently before this PR is mergeable.

Please review before merging.

## Release Notes

* **New Features**
  * Added a floating Copilot Chat panel to the application interface
  * Chat widget displays message history, suggested prompts, and user-assistant interactions
  * Supports context awareness and displays tool interactions with their results
  * Includes loading states and error handling for messaging functionality

* **Documentation**
  * Added implementation plan documenting the Copilot Chat feature architecture and acceptance criteria

### CODE REVIEW

File: .plans/issue-509.md [hunk 1]
```diff
@@ -0,0 +1,51 @@
+No page store exists. The dev-plan board explicitly notes this may be stubbed. I now have enough to write the plan. Here's the output:
```
Comment (yihua-zhuo) [inline .plans/issue-509.md:1]: **WARNING** (suggestion)

**Issue:** Stale meta-narrative text ('No page store exists...I now have enough to write the plan.') was not cleaned up before committing the plan file. (Rule 140: plan must be re-checked against the issue)

**Suggestion:** Delete line 1 and start the file directly with the plan heading.

File: .plans/issue-509.md [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
...ntext bar; `useSendCopilotMessage` for the API call. The FAB button is fixed-position bottom-right (`fixed bottom-6 right-6 z-[9999]` per dev-plan §4.3). The chat window is a panel anchored above the FAB (`bottom-24 right-6`) with the same width/height pattern as the existing `AIPanel` in `frontend/src/lib/components/ai-panel.tsx` L82-L87. The context bar at the top shows `"Chatting about: Customer #{N}"` when `customer_id` is set, otherwise `"No context"` (also reflects `opportunity_id` when set). The message list renders user messages right-aligned (primary background) and assistant messages left-aligned (muted background). Suggested prompts (e.g. "Show my top leads", "Summarize this ticket", "What's new today?") are shown only when `messages.length === 0`. On send: append the user message, call `mutate`, on success append the assistant response, clear the input. On error: append a red inline error message ("Copilot unavailable") — matches dev-plan §7 fallback. For tool-call rendering: when a message's response includes non-empty `tool_calls`, render each as a structured card (bordered, rounded, with tool name and a summary of arguments/result) below the assistant message bubble.
```
Comment (yihua-zhuo) [inline .plans/issue-509.md:28]: **INFO** (suggestion)

**Issue:** Off-by-one: plan says AIPanel is at L79 of layout.tsx, but the actual import and render is at L80. (Rule 140: stale file references)

**Suggestion:** Update L28 and L30 of the plan to reference L80 for AIPanel.

File: frontend/src/app/(app)/layout.tsx [hunk 3]
```diff
@@ -77,6 +78,7 @@ export default function AppLayout({ children }: { children: React.ReactNode }) {
       </div>
 
       <AIPanel />
```
Comment (yihua-zhuo) [inline frontend/src/app/(app)/layout.tsx:80]: **INFO** (suggestion)

**Issue:** Both `<AIPanel />` and `<CopilotChat />` are rendered as global floating panels; if AIPanel also uses bottom-right fixed positioning the FABs may overlap.

**Suggestion:** Verify AIPanel's position; consider offsetting one FAB (e.g. `bottom-6 right-24`) or consolidating them as the plan's risk #3 notes.

File: frontend/src/components/CopilotChat.tsx [hunk 4]
```diff
@@ -0,0 +1,234 @@
...
+                </div>
+                <div className="flex items-center gap-1 rounded-xl bg-muted px-3 py-2">
+                  <div className="h-1.5 w-1.5 rounded-full bg-muted-foreground animate-bounce [animation-delay:0ms]" />
+                  <div className="h-1.5 w-1.5 rounded-full bg-muted-foreground animate-bounce [animation-delay:150ms]" />
+                  <div className="h-1.5 w-1.5 rounded-full bg-muted-foreground animate-bounce [animation-delay:300ms]" />
+                </div>
+              </div>
+            )}
+
+            {error && messages.length === 0 && (
```
Comment (yihua-zhuo) [inline frontend/src/components/CopilotChat.tsx]: **WARNING** (suggestion)

**Issue:** The `error && messages.length === 0` block is effectively dead code: `handleSend` always appends a user message before calling `mutate`, so `messages.length` is never 0 when `error` becomes truthy. (Rule 17: dead code removed)

**Suggestion:** Remove the block, or change the condition to display the error alongside the 'Copilot unavailable' assistant message already injected by the onError callback.

File: frontend/src/components/CopilotChat.tsx [hunk 5]
```diff
@@ -0,0 +1,234 @@
...
+          ...prev,
+          {
+            id: crypto.randomUUID(),
+            role: "assistant",
+            content: data.response,
+            toolCalls: data.tool_calls?.length ? data.tool_calls : undefined,
+          },
+        ]);
+      },
+      onError: () => {
```
Comment (yihua-zhuo) [inline frontend/src/components/CopilotChat.tsx]: **INFO** (suggestion)

**Issue:** The `onError` callback discards the error parameter without logging it, making API failures invisible in browser console. (Rule 41: errors logged with useful context)

**Suggestion:** Change the signature to `onError: (err) => { console.error('Copilot request failed', err); ... }` for debugging.

File: frontend/src/components/CopilotChat.tsx [hunk 6]
```diff
@@ -0,0 +1,234 @@
...
+                    {m.content}
+                  </div>
+                  {m.role === "user" && (
+                    <div className="mt-0.5 flex h-6 w-6 flex-shrink-0 items-center justify-center rounded-full bg-primary/10">
+                      <User className="h-3.5 w-3.5 text-primary" />
+                    </div>
+                  )}
+                </div>
+                {/* Tool-call cards */}
+                {m.toolCalls?.map((tc, i) => (
```
Comment (yihua-zhuo) [inline frontend/src/components/CopilotChat.tsx]: **INFO** (suggestion)

**Issue:** Tool-call cards use array index as `key`, which is acceptable here since the list is append-only and never reordered, but would break if tool_calls are ever mutated in place. (Rule 15: avoids unnecessary cleverness — index keys are safe but fragile)

**Suggestion:** If the backend ever returns stable tool-call IDs, use those as keys instead of the index.

File: frontend/src/hooks/usePageContext.ts [hunk 7]
```diff
@@ -0,0 +1,7 @@
+export function usePageContext(): {
+  customer_id: number | null;
+  opportunity_id: number | null;
+} {
+  // TODO: wire to page store once one exists
```
Comment (yihua-zhuo) [inline frontend/src/hooks/usePageContext.ts]: **INFO** (suggestion)

**Issue:** TODO comment is appropriate per the plan, but will silently return nulls if the page store is never built, leaving the context bar permanently showing 'No context'.

**Suggestion:** Consider adding a log warning on first call, or a dev-only assertion, so the stub is visible during development.

File: frontend/src/lib/api/queries.ts [hunk 8]
```diff
@@ -566,3 +566,23 @@ export function useAutomationLogs(page = 1, pageSize = 20, ruleId?: number, stat
...
+interface CopilotChatResponse {
+  response: string;
+  conversation_id: number;
+  tool_calls: Array<{ tool: string; arguments: Record<string, unknown>; result?: unknown }>;
+}
+
+export function useSendCopilotMessage() {
+  const token = useAuthStore((s) => s.token);
+  return useMutation({
+    mutationFn: (message: string) =>
```
Comment (yihua-zhuo) [inline frontend/src/lib/api/queries.ts:581]: **WARNING** (suggestion)

**Issue:** useSendCopilotMessage puts the user message in the query string (?message=...) instead of the request body — long messages will exceed URL length limits and break the request (Rule 92: proper request design).

**Suggestion:** apiClient.post(`/api/v1/copilot/chat`, { message }, token ?? undefined)

**Comment by yihua-zhuo** [comment]
## Claude Code Review (inline)

**Status:** `fail`
**Inline comments posted:** 6

**Summary:** Reviewed 5 file(s) between origin/master and origin/auto/issue-509.

### Issues without a usable file/line
- `.plans/issue-509.md`: Plan says FAB at `bottom-6 right-6` in Step 3 but actual implementation uses `right-24`; plan is stale and contradicts itself — violates Rule 140 (plan must be re-checked against implementation).
- `.plans/issue-509.md`: Plan says chat window at `bottom-24 right-6` in Step 3 but implementation uses `right-24` — stale requirement that would cause overlap with AIPanel if followed literally.
- `.plans/issue-509.md`: Plan acknowledges no page store exists and usePageContext returns nulls, meaning the context bar will always show 'No context' — a known degradation silently accepted without remediation path.
- `frontend/src/components/CopilotChat.tsx`: No conversation_id is sent to or tracked from the backend — the backend cannot maintain conversation state, so every message is treated as a new independent query.
- `frontend/src/hooks/usePageContext.ts`: Named `usePageContext` but contains no React hooks (no useState, useContext, useEffect) — violates React naming convention that 'use' prefix indicates a hook. This is a plain function masquerading as a hook.
- `.plans/issue-509.md`: Test plan section states 'no unit tests' for a 230-line component with API calls, state management, error handling, and tool-call rendering — violates Rules 61, 63, 64, 65 (unit tests, determinism, edge cases, failure scenarios).
- `.plans/issue-509.md`: Plan does not address the UX concern of having two floating panels (AIPanel + CopilotChat) both mounted globally in the layout — two 420px×600px panels is cluttering.
- `.plans/issue-509.md`: Acceptance criteria do not verify that tool-call cards render correctly or that error states display — only checks tsc, lint, build, file existence, and manual E2E.
- `.plans/issue-509.md`: Plan references dev-plan §4.3 for FAB positioning and §4.4 for risks but does not document the right-24 deviation inline, making future maintenance harder.
- `frontend/src/app/(app)/layout.tsx`: No tests exist for the layout component despite it being the mount point for two major floating panels and global auth guards.
- `frontend/src/app/(app)/layout.tsx`: z-index values are inconsistent: AIPanel FAB is z-40, AIPanel panel is z-50, CopilotChat is z-[9999] for both — CopilotChat will always overlay AIPanel regardless of which was opened first.
- `frontend/src/components/CopilotChat.tsx`: No unit tests for this 230-line component with API calls, state management, error handling, and tool-call rendering — violates Rules 61, 63, 64, 65.
- `frontend/src/components/CopilotChat.tsx`: No accessibility features: no aria-live region for new messages, no keyboard shortcut to close the panel (e.g. Escape), no focus trap when the chat is open.
- `frontend/src/components/CopilotChat.tsx`: No rate limiting or debounce on the send button — a user could rapidly click send and fire multiple API calls.
- `frontend/src/components/CopilotChat.tsx`: No cleanup of pending state if the component unmounts mid-request — React state updates on an unmounted component will trigger warnings.
- `frontend/src/components/CopilotChat.tsx`: The FAB trigger uses a raw <button> (L82) while the close button uses the shared <Button> component (L99) — inconsistent button usage within the same component.
- `frontend/src/hooks/usePageContext.ts`: console.warn fires on every call in non-production — a hook that may be called many times per render will produce excessive log noise.
- `frontend/src/hooks/usePageContext.ts`: No tests for this stub — even a trivial stub should have a test asserting the return shape so future changes don't silently break consumers.

### Issues that could not be posted inline
`frontend/src/app/(app)/layout.tsx` L89: 
**CRITICAL** (critical)

**Issue:** The onSubmit handler wraps the mutation in a try/catch with an empty catch block that silently swallows errors — violates Rule 46 (no silent exception swallowing). The comment claims the mutation hook handles errors, but no toast or user feedback is wired up.

**Suggestion:** Remove the try/catch and let the error propagate to React Query's onError handler, or add a visible error notification (e.g. toast) in the catch block. — _HTTP 422: {"message":"Validation Failed","errors":[{"resource":"PullRequestReviewComment","code":"custom","field":"pull_request_review_thread.line","message":"could not be resolved"}],"documentation_url":"https://docs.github.com/rest/pulls/comments#create-a-review-comment-for-a-pull-request","status":"422"}_
`frontend/src/lib/api/queries.ts` L15: 
**CRITICAL** (critical)

**Issue:** Rule 147: `qk.tasks(page, status)` key omits `priority`, `assigned_to`, `createdAfter`, and `createdBefore` which are all sent as query parameters — changing any of those filters within the same page+status returns stale cached data.

**Suggestion:** Change to `tasks: (page = 1, status = "", priority = "", assignedTo = "", createdAfter = "", createdBefore = "") => ["tasks", page, status, priority, assignedTo, createdAfter, createdBefore] as const`. — _HTTP 422: {"message":"Validation Failed","errors":[{"resource":"PullRequestReviewComment","code":"custom","field":"pull_request_review_thread.line","message":"could not be resolved"}],"documentation_url":"https://docs.github.com/rest/pulls/comments#create-a-review-comment-for-a-pull-request","status":"422"}_
`frontend/src/lib/api/queries.ts` L82: 
**CRITICAL** (critical)

**Issue:** Detail endpoint `/api/v1/customers/{id}` is typed as `ApiEnvelope<PaginatedResponse<T>>` but the file's own comment (lines 42–43) notes this is wrong for detail fetches — the server returns the record directly via `data: T`, not `data: { items: [...] }`.

**Suggestion:** Use `ApiEnvelopeSingle<Record<string, unknown>>` instead of `ApiEnvelope<Record<string, unknown>>`. — _HTTP 422: {"message":"Validation Failed","errors":[{"resource":"PullRequestReviewComment","code":"custom","field":"pull_request_review_thread.line","message":"could not be resolved"}],"documentation_url":"https://docs.github.com/rest/pulls/comments#create-a-review-comment-for-a-pull-request","status":"422"}_
`frontend/src/lib/api/queries.ts` L181: 
**CRITICAL** (critical)

**Issue:** Detail endpoint `/api/v1/tickets/{id}` is typed as `ApiEnvelope<PaginatedResponse<T>>` but returns a single record — `data.items` will be undefined at runtime.

**Suggestion:** Use `ApiEnvelopeSingle<Record<string, unknown>>`. — _HTTP 422: {"message":"Validation Failed","errors":[{"resource":"PullRequestReviewComment","code":"custom","field":"pull_request_review_thread.line","message":"could not be resolved"}],"documentation_url":"https://docs.github.com/rest/pulls/comments#create-a-review-comment-for-a-pull-request","status":"422"}_
`frontend/src/lib/api/queries.ts` L350: 
**CRITICAL** (critical)

**Issue:** Detail endpoint `/api/v1/tasks/{id}` is typed as `ApiEnvelope<PaginatedResponse<T>>` but returns a single record.

**Suggestion:** Use `ApiEnvelopeSingle<Record<string, unknown>>`. — _HTTP 422: {"message":"Validation Failed","errors":[{"resource":"PullRequestReviewComment","code":"custom","field":"pull_request_review_thread.line","message":"could not be resolved"}],"documentation_url":"https://docs.github.com/rest/pulls/comments#create-a-review-comment-for-a-pull-request","status":"422"}_
`frontend/src/lib/api/queries.ts` L7: 
**WARNING** (suggestion)

**Issue:** Rule 147: `qk.customers(page)` query key omits `pageSize`, but `useCustomers` sends `page_size` as a request parameter — different pageSize values with the same page share the same cache key, returning stale data.

**Suggestion:** Change to `customers: (page = 1, pageSize = 20) => ["customers", page, pageSize] as const` and pass both args from the hook. — _HTTP 422: {"message":"Validation Failed","errors":[{"resource":"PullRequestReviewComment","code":"custom","field":"pull_request_review_thread.line","message":"could not be resolved"}],"documentation_url":"https://docs.github.com/rest/pulls/comments#create-a-review-comment-for-a-pull-request","status":"422"}_
`frontend/src/lib/api/queries.ts` L11: 
**WARNING** (suggestion)

**Issue:** Rule 147: `qk.tickets(page, status)` query key omits `pageSize`, but the request sends it — different pageSize values with the same page+status will collide in the cache.

**Suggestion:** Change to `tickets: (page = 1, pageSize = 20, status = "") => ["tickets", page, pageSize, status] as const`. — _HTTP 422: {"message":"Validation Failed","errors":[{"resource":"PullRequestReviewComment","code":"custom","field":"pull_request_review_thread.line","message":"could not be resolved"}],"documentation_url":"https://docs.github.com/rest/pulls/comments#create-a-review-comment-for-a-pull-request","status":"422"}_
`frontend/src/lib/api/queries.ts` L18: 
**WARNING** (suggestion)

**Issue:** Rule 147: `qk.notifications(page, unreadOnly)` omits `pageSize` but the request sends it — cache collisions across different page sizes.

**Suggestion:** Change to `notifications: (page = 1, pageSize = 20, unreadOnly = false) => ["notifications", page, pageSize, unreadOnly] as const`. — _HTTP 422: {"message":"Validation Failed","errors":[{"resource":"PullRequestReviewComment","code":"custom","field":"pull_request_review_thread.line","message":"could not be resolved"}],"documentation_url":"https://docs.github.com/rest/pulls/comments#create-a-review-comment-for-a-pull-request","status":"422"}_
`frontend/src/lib/api/queries.ts` L20: 
**WARNING** (suggestion)

**Issue:** Rule 147: `qk.activities(page, type)` omits `pageSize` but the request sends it — cache collisions across different page sizes.

**Suggestion:** Change to `activities: (page = 1, pageSize = 20, type = "") => ["activities", page, pageSize, type] as const`. — _HTTP 422: {"message":"Validation Failed","errors":[{"resource":"PullRequestReviewComment","code":"custom","field":"pull_request_review_thread.line","message":"could not be resolved"}],"documentation_url":"https://docs.github.com/rest/pulls/comments#create-a-review-comment-for-a-pull-request","status":"422"}_
`frontend/src/lib/api/queries.ts` L240: 
**INFO** (suggestion)

**Issue:** Rule 147: `useChangeTicketStatus` invalidates the ticket list and detail but does not invalidate `qk.ticketActivity(ticketId)` — a status change is typically recorded in the activity log, making the activity cache stale.

**Suggestion:** Add `qc.invalidateQueries({ queryKey: qk.ticketActivity(vars.ticketId) });` to the onSuccess handler. — _HTTP 422: {"message":"Validation Failed","errors":[{"resource":"PullRequestReviewComment","code":"custom","field":"pull_request_review_thread.line","message":"could not be resolved"}],"documentation_url":"https://docs.github.com/rest/pulls/comments#create-a-review-comment-for-a-pull-request","status":"422"}_
`frontend/src/lib/api/queries.ts` L252: 
**WARNING** (suggestion)

**Issue:** Rule 147: `useBulkUpdateTickets` does not invalidate individual ticket detail keys for any of the tickets in `ticket_ids` — the detail cache for each affected ticket will be stale.

**Suggestion:** In onSuccess, iterate over the `ticket_ids` from the mutation variables and invalidate each `qk.ticket(id)` key. — _HTTP 422: {"message":"Validation Failed","errors":[{"resource":"PullRequestReviewComment","code":"custom","field":"pull_request_review_thread.line","message":"could not be resolved"}],"documentation_url":"https://docs.github.com/rest/pulls/comments#create-a-review-comment-for-a-pull-request","status":"422"}_
`frontend/src/lib/api/queries.ts` L381: 
**WARNING** (suggestion)

**Issue:** Rule 147: `useCompleteTask` invalidates only `["tasks"]` (list) but not the specific task detail `qk.task(id)` — detail view will show stale status.

**Suggestion:** In onSuccess, also call `qc.invalidateQueries({ queryKey: qk.task(id) });` using the mutation's `id` variable. — _HTTP 422: {"message":"Validation Failed","errors":[{"resource":"PullRequestReviewComment","code":"custom","field":"pull_request_review_thread.line","message":"could not be resolved"}],"documentation_url":"https://docs.github.com/rest/pulls/comments#create-a-review-comment-for-a-pull-request","status":"422"}_
`frontend/src/lib/api/queries.ts` L391: 
**WARNING** (suggestion)

**Issue:** Rule 147: `useDeleteTask` invalidates only `["tasks"]` (list) but not the specific task detail `qk.task(id)`.

**Suggestion:** In onSuccess, also call `qc.invalidateQueries({ queryKey: qk.task(id) });` using the mutation's `id` variable. — _HTTP 422: {"message":"Validation Failed","errors":[{"resource":"PullRequestReviewComment","code":"custom","field":"pull_request_review_thread.line","message":"could not be resolved"}],"documentation_url":"https://docs.github.com/rest/pulls/comments#create-a-review-comment-for-a-pull-request","status":"422"}_
`frontend/src/lib/api/queries.ts` L525: 
**WARNING** (suggestion)

**Issue:** Rule 147: `useUpdateAutomationRule` receives `id` but does not invalidate `qk.automationRule(id)` — the detail view for that rule will be stale.

**Suggestion:** Add `qc.invalidateQueries({ queryKey: qk.automationRule(vars.id) });` to onSuccess. — _HTTP 422: {"message":"Validation Failed","errors":[{"resource":"PullRequestReviewComment","code":"custom","field":"pull_request_review_thread.line","message":"could not be resolved"}],"documentation_url":"https://docs.github.com/rest/pulls/comments#create-a-review-comment-for-a-pull-request","status":"422"}_
`frontend/src/lib/api/queries.ts` L551: 
**WARNING** (suggestion)

**Issue:** Rule 147: `useToggleAutomationRule` receives `id` but does not invalidate `qk.automationRule(id)` — toggling changes the rule's enabled state visible in the detail view.

**Suggestion:** Add `qc.invalidateQueries({ queryKey: qk.automationRule(id) });` to onSuccess. — _HTTP 422: {"message":"Validation Failed","errors":[{"resource":"PullRequestReviewComment","code":"custom","field":"pull_request_review_thread.line","message":"could not be resolved"}],"documentation_url":"https://docs.github.com/rest/pulls/comments#create-a-review-comment-for-a-pull-request","status":"422"}_
`frontend/src/lib/api/queries.ts` L61: 
**WARNING** (suggestion)

**Issue:** Rule 148: `useCurrentUser` bypasses the shared `apiClient` and `token ?? undefined` pattern used by every other hook — instead it dynamically imports `../api/auth` and passes `token ?? ""` to `getMe`, duplicating auth logic.

**Suggestion:** Use `apiClient.get<ApiEnvelopeSingle<UserSummary>>("/api/v1/auth/me", token ?? undefined)` consistent with the other hooks. — _HTTP 422: {"message":"Validation Failed","errors":[{"resource":"PullRequestReviewComment","code":"custom","field":"pull_request_review_thread.line","message":"could not be resolved"}],"documentation_url":"https://docs.github.com/rest/pulls/comments#create-a-review-comment-for-a-pull-request","status":"422"}_
`frontend/src/lib/api/queries.ts` L90: 
**INFO** (suggestion)

**Issue:** Consistency: `useSearchCustomers` uses a raw `["customers", "search", keyword]` array instead of the `qk` factory pattern used everywhere else.

**Suggestion:** Add `customersSearch: (keyword: string) => ["customers", "search", keyword] as const` to the `qk` object and use `qk.customersSearch(keyword)`. — _HTTP 422: {"message":"Validation Failed","errors":[{"resource":"PullRequestReviewComment","code":"custom","field":"pull_request_review_thread.line","message":"could not be resolved"}],"documentation_url":"https://docs.github.com/rest/pulls/comments#create-a-review-comment-for-a-pull-request","status":"422"}_
`frontend/src/lib/api/queries.ts` L410: 
**INFO** (suggestion)

**Issue:** Consistency: `useActivity` uses a raw `["activity", id]` array instead of the `qk` factory, and no `qk.activity` factory exists — all other query keys go through `qk`.

**Suggestion:** Add `activity: (id: number) => ["activity", id] as const` to `qk` and use `qk.activity(id)`. — _HTTP 422: {"message":"Validation Failed","errors":[{"resource":"PullRequestReviewComment","code":"custom","field":"pull_request_review_thread.line","message":"could not be resolved"}],"documentation_url":"https://docs.github.com/rest/pulls/comments#create-a-review-comment-for-a-pull-request","status":"422"}_

File: frontend/src/components/CopilotChat.tsx [hunk 9]
```diff
@@ -0,0 +1,229 @@
...
+    });
+  }
+
+  return (
+    <>
+      {/* FAB trigger — offset to the left of AIPanel's FAB (bottom-6 right-6) to avoid overlap */}
+      <button
+        type="button"
+        onClick={() => setIsOpen(!isOpen)}
+        className="fixed bottom-6 right-24 z-[9999] flex h-12 w-12 items-center justify-center rounded-full bg-primary text-primary-foreground shadow-lg hover:bg-primary/90 transition-colors"
```
Comment (yihua-zhuo) [inline frontend/src/components/CopilotChat.tsx:85]: **CRITICAL** (critical)

**Issue:** FAB at `bottom-6 right-24 z-[9999]` offsets from AIPanel's right-6 but the plan specified right-6 — the plan and implementation disagree, and the high z-[9999] will always overlay AIPanel's z-50 panel.

**Suggestion:** Use a consistent z-index (e.g. z-50) and coordinate with AIPanel so only one panel is visible at a time.

File: frontend/src/components/CopilotChat.tsx [hunk 10]
```diff
@@ -0,0 +1,229 @@
...
+          ...prev,
+          {
+            id: crypto.randomUUID(),
+            role: "assistant",
+            content: data.response,
+            toolCalls: data.tool_calls?.length ? data.tool_calls : undefined,
+          },
+        ]);
+      },
+      onError: (err) => {
```
Comment (yihua-zhuo) [inline frontend/src/components/CopilotChat.tsx:65]: **CRITICAL** (critical)

**Issue:** handleSend error handler appends an assistant message with 'Copilot unavailable' and only console.error — no user-visible error indication beyond the text, and the error object is not fully handled — violates Rule 46.

**Suggestion:** Show a distinct error state (e.g. red border, retry button) and log the full error. Use mutateAsync to await the response and handle errors explicitly.

File: frontend/src/app/(app)/layout.tsx [hunk 11]
```diff
@@ -77,6 +78,7 @@ export default function AppLayout({ children }: { children: React.ReactNode }) {
...
+      <CopilotChat />
```
Comment (yihua-zhuo) [inline frontend/src/app/(app)/layout.tsx:81]: **WARNING** (suggestion)

**Issue:** CopilotChat is mounted globally for all authenticated pages including settings/admin — may be unintended exposure of AI features on every page.

**Suggestion:** Restrict CopilotChat to specific route groups or add a feature flag to disable it on admin/settings pages.

File: frontend/src/components/CopilotChat.tsx [hunk 12]
```diff
@@ -0,0 +1,229 @@
...
+                      <Wrench className="h-3 w-3 text-muted-foreground" />
+                      {tc.tool}
+                    </div>
+                    <div className="text-muted-foreground">
+                      {Object.keys(tc.arguments).length > 0 && (
+                        <span>Args: {JSON.stringify(tc.arguments)}</span>
+                      )}
+                    </div>
+                    {tc.result != null && (
+                      <div className="text-muted-foreground truncate">
```
Comment (yihua-zhuo) [inline frontend/src/components/CopilotChat.tsx:178]: **WARNING** (suggestion)

**Issue:** JSON.stringify(tc.result) is rendered without truncation in the tool-call card — a large result object can make the card very tall and break the fixed-size chat window layout.

**Suggestion:** Truncate the result to a reasonable length (e.g. 500 chars) with a 'Show more' toggle, or render it in a scrollable container with max-height.

File: frontend/src/components/CopilotChat.tsx [hunk 13]
```diff
@@ -0,0 +1,229 @@
...
+        ? `Chatting about: Opportunity #${opportunity_id}`
+        : "No context";
+
+  function handleSend(text?: string) {
+    const msg = (text ?? input).trim();
+    if (!msg || isPending) return;
+    setInput("");
+    setMessages((prev) => [
+      ...prev,
+      { id: crypto.randomUUID(), role: "user", content: msg },
```
Comment (yihua-zhuo) [inline frontend/src/components/CopilotChat.tsx:49]: **INFO** (suggestion)

**Issue:** Message IDs use crypto.randomUUID() which is fine for modern browsers but not available in older environments — React 18+ already requires modern browsers, so this is acceptable but worth noting.

**Suggestion:** No action needed if browser support targets are modern (Chrome 92+, Firefox 95+, Safari 15.4+). If older support is needed, use a UUID polyfill or nanoid.
